File: src/main/kotlin/ru/tikhonovdo/enrichment/config/DataSourceConfig.java
package ru.tikhonovdo.enrichment.config;

import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

@Configuration
public class DataSourceConfig {

//    private final static String PRIMARY_DS = "primaryDataSource";
    private final static String PRIMARY_PROPERTIES = "primaryProperties";

    @Primary
    @Bean(PRIMARY_PROPERTIES)
    @ConfigurationProperties(prefix = "spring.datasource")
    public DataSourceProperties primaryDataSourceProperties() {
        return new DataSourceProperties();
    }
}

File: src/main/kotlin/ru/tikhonovdo/enrichment/config/DataSourceConfig.kt
package ru.tikhonovdo.enrichment.config

import com.zaxxer.hikari.HikariDataSource
import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
import javax.sql.DataSource

@Configuration
class DataSourceConfig {

    @Primary
    @Bean
    @ConfigurationProperties(prefix = "spring.datasource")
    fun dataSourceProperties(): DataSourceProperties = DataSourceProperties()

    @Primary
    @Bean
    @ConfigurationProperties(prefix = "spring.datasource.hikari")
    fun dataSource(dataSourceProperties: DataSourceProperties): DataSource {
        return dataSourceProperties.initializeDataSourceBuilder().type(HikariDataSource::class.java).build()
    }

    @Primary
    @Bean(name = ["jdbcPrimary"])
    fun primaryJdbcTemplate(dataSource: DataSource) = JdbcTemplate(dataSource)

    @Bean
    @Primary
    fun namedParameterJdbcTemplate(jdbcTemplate: JdbcTemplate): NamedParameterJdbcTemplate? {
        return NamedParameterJdbcTemplate(jdbcTemplate)
    }
}